llvm/lib/CodeGen/SelectionDAG/SelectionDAG.cpp

```diff
@@ -11713,7 +11714,10 @@ void SelectionDAG::CreateTopologicalOrder(std::vector<SDNode *> &Order) {
 }
 
 #ifndef NDEBUG
-void SelectionDAG::VerifyDAGDivergence() {
+void SelectionDAG::VerifyDAGDivergence(TargetTransformInfo *TTI) {
+  if (TTI && !TTI->hasBranchDivergence())
```

Comment:
I'm pretty sure it's impossible for TTI to be null. It should always be an available analysis, and the member is not declared for !NDEBUG

Reply:
Although with the usage maybe this field should just be replaced with the bool HasBranchDivergence?

Reply:
> I'm pretty sure it's impossible for TTI to be null.

TTI is only set inside `LLVM_ENABLE_ABI_BREAKING_CHECKS`.

Reply:
> Although with the usage maybe this field should just be replaced with the bool HasBranchDivergence?

Yeah that might be cleaner.